src/OSPSuite.Core/Domain/Services/ParameterValuesCreator.cs

```diff
@@ -85,18 +89,72 @@ public ParameterValue CreateParameterValue(ObjectPath parameterPath, double valu
 
-      private IEnumerable<ParameterValue> createFrom(IContainer container, IReadOnlyList<MoleculeBuilder> molecules, Func<IParameter, IContainer, bool> createFor) =>
-         molecules.SelectMany(x => createFrom(container, x, createFor));
+      private IEnumerable<ParameterValue> createExpressionFrom(IContainer container, IReadOnlyList<MoleculeBuilder> molecules) => molecules.SelectMany(x => createExpressionFrom(container, x));
 
-      private IEnumerable<ParameterValue> createFrom(IContainer container, MoleculeBuilder molecule, Func<IParameter, IContainer, bool> createFor) =>
-         molecule.Parameters.Where(x => createFor(x, container)).Select(x => CreateParameterValue(objectPathForParameterInContainer(container, x.Name, molecule.Name), x));
+      private void initializeFormulasFor(IReadOnlyList<ParameterValue> parameterValues, IReadOnlyList<ExpressionParameter> expressionParameters)
+      {
+         var formulaCache = new FormulaCache();
+         parameterValues.Each(formulaTarget =>
+         {
+            var formulaSource = formulaSourceFor(expressionParameters.AllByName(formulaTarget.Name).ToList(), formulaTarget);
+
+            if (formulaSource != null)
+               formulaTarget.Formula = _cloneManager.Clone(formulaSource.Formula, formulaCache);
+         });
+      }
+
+      private bool shouldCloneFormulaFor(ParameterValue parameterValue) => parameterValue.Formula == null || parameterValue.Formula.IsConstant();
+
+      private static ExpressionParameter formulaSourceFor(List<ExpressionParameter> nameMatchedExpressionParameters, ParameterValue formulaTarget)
+      {
+         var formulaSource = pathMatchedExpressionParameterFor(nameMatchedExpressionParameters, formulaTarget);
+
+         if (formulaSource == null && hasCompartment(formulaTarget))
+            formulaSource = compartmentMatchedExpressionParameterFor(formulaTarget, nameMatchedExpressionParameters);
+
+         return formulaSource;
+      }
+
+      private static ExpressionParameter compartmentMatchedExpressionParameterFor(ParameterValue formulaTarget, List<ExpressionParameter> nameMatchedExpressionParameters) =>
+         nameMatchedExpressionParameters.Where(hasCompartment).FirstOrDefault(x => Equals(compartmentFor(x), compartmentFor(formulaTarget)));
+
+      private static string compartmentFor(ParameterValue formulaTarget) => formulaTarget.Path[compartmentIndex(formulaTarget)];
+
+      private static int compartmentIndex(PathAndValueEntity x) => x.Path.Count - 3;
+
+      private static ExpressionParameter pathMatchedExpressionParameterFor(List<ExpressionParameter> nameMatchedExpressionParameters, ParameterValue formulaTarget) =>
+         nameMatchedExpressionParameters.FirstOrDefault(x => Equals(x.Path, formulaTarget.Path));
+
+      private static bool hasCompartment(ParameterValue formulaTarget) => formulaTarget.Path.Count >= 5;
```

Comment:
uff
That's hard to understand and requires at least a comment what 5 means (ideally with a path example).

Reply:
And we have compartments on different levels, so hmm...
But waiting for the comment first :)

Reply:
Ok, I commented these. This seems to be by convention. We use this kind of convention elsewhere for Expression parameters. But, maybe 5 is not the right number. 

Eg. if we have a path with only 3 elements, then we could take the third-last as compartment maybe. That would be the first element.

Reply:
yeah, for example t 1/2 (liver) would return false. 

![image](https://github.com/user-attachments/assets/9df24659-b32b-4dd5-a9fa-2be1b1ceadd1)